### PR TITLE
Export types

### PR DESCRIPTION
**Pull request by runcom**
/cc @vbatts @stevvooe @philips 

This is another version https://github.com/opencontainers/image-spec/pull/147 w/o using any code generation tool - as discussed over there

Close https://github.com/opencontainers/image-spec/pull/147

### CODE REVIEW

File: specs-go/versioned.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package specs
+
+// Versioned provides a struct with the manifest schemaVersion and . Incoming
```
Comment (wking) [inline specs-go/versioned.go]: “and . Incoming” is probably a typo.  Maybe you meant “schemaVersion and mediaType.  Incoming”?

**Comment by vbatts** [comment]
as a first pass, this LGTM

**Comment by stevvooe** [comment]
LGTM on first pass.

**Comment by vbatts** [comment]
cool. i'll merge this and we can iterate on it.

**Pull request merged**